SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,256 @@
+  if constexpr (Alignment == alignment_type::element) {
+    return element_aligned;
+  } else if constexpr (Alignment == alignment_type::vector) {
+    return vector_aligned;
+  } else if constexpr (Alignment == alignment_type::overaligned) {
+    return overaligned<alignof(std::max_align_t)>;
+  } else {
+    static_assert(Alignment != Alignment,
+                  "Unexpected alignment_type enum field");
+  }
+}
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T, typename Alignment>
+struct Kernel1;
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename DataT, int NumElems, typename TestCaseT, typename AlignmentT>
+class run_test {
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+    const std::vector<DataT> ref_data = generate_ref_data<DataT, NumElems>();
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    const std::size_t alignment_value =
+        get_alignment_value<DataT, NumElems, AlignmentT::value>();
+    const std::size_t container_extra_size =
+        alignment_value / sizeof(DataT) + 1;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(NumElems + container_extra_size,
+                                         shared_allocator<DataT>(queue));
+
+    size_t object_size = NumElems * sizeof(DataT);
+    size_t buffer_size = object_size + alignment_value;
+
+    // Using pointer to next element to avoid bigger alignment of USM shared
+    // memory to guarantee the weakest alignment. By thi std::align can provide
+    // expected alignment on the small values of an alignment.
+    void *ref = shared_ref_data.data() + 1;
+    if (std::align(alignment_value, object_size, ref, buffer_size) == nullptr) {
+      return false;
+    }
+    DataT *const ref_aligned = static_cast<DataT *>(ref);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_aligned[i] = ref_data[i];
+    }
+
+    auto alignment = get_alignment<DataT, NumElems, AlignmentT::value>();
+    queue.submit([&](sycl::handler &cgh) {
```

Comment:
Maybe it's better to have it on device side to avoid any unexpected problems? Something like
```
          [=]() SYCL_ESIMD_KERNEL {
            const auto alignment = get_alignment<DataT, NumElems, AlignmentT::value>();
            TestCaseT::template call_simd_ctor<DataT, NumElems>(ref_aligned,
                                                                out, alignment);
          });
```

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/769/commits/9de67432a3606acda7db15caab8cc4c7ef03c3f4#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7R192-R196)